Sources/NavigatorKit/Navigator.swift

```diff
@@ -51,11 +54,15 @@ public class Navigator: ObservableObject {
 
         stack[stack.count - 1].links[destination]?.wrappedValue = true
         stack[stack.count - 1].args[destination] = args
+        self.completion = completion
```

Comment:
I believe this will have issues if you have multiple levels of screens requiring values to be passed back. For instance, if screen 1 launches screen 2 and needs values, but then screen 2 launches screen 3 for values, I don't think the first screen's completion will ever get called.

Reply:
Ah yes that's a good point. Thanks! I was able to reproduce this exact issue. I'll iterate on what I have and update this PR